packages/@pollyjs/core/src/utils/merge-configs.js

```diff
@@ -1,4 +1,32 @@
 import mergeWith from 'lodash-es/mergeWith';
+import { assert, EXPIRY_STRATEGIES } from '@pollyjs/utils';
+
+function deprecateRecordIfExpired(mergedConfig) {
+  const bothAreUsed =
+    mergedConfig.hasOwnProperty('recordIfExpired') &&
+    mergedConfig.hasOwnProperty('expiryStrategy');
+
+  // throw if you used both recordIfExpired and expiryStrategy
+  assert(
+    'recordIfExpired is deprecated and cannot be used with expiryStrategy',
+    !bothAreUsed
+  );
+
+  // return if you only used expiryStrategy
+  if (mergedConfig.hasOwnProperty('expiryStrategy')) {
+    return mergedConfig;
+  }
+
+  if (mergedConfig.recordIfExpired) {
+    // replace recordIfExpired: true with expiryStrategy: record
+    mergedConfig.expiryStrategy = EXPIRY_STRATEGIES.RECORD;
+  } else {
+    // replace recordIfExpired: false with expiryStrategy: warn
+    mergedConfig.expiryStrategy = EXPIRY_STRATEGIES.WARN;
+  }
+
+  return mergedConfig;
```

Comment:
We should delete the `recordIfExpired` option at the end here. No need to keep it around 😄

Reply:
done!